### PR TITLE
Road 134 - Add the proper permissions to the RS quicksuggest table

### PR DESCRIPTION
**Pull request by Dexterp37**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/search_derived/remotesettings_suggestions_v1/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+workgroup_access:
```
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/search_derived/remotesettings_suggestions_v1/metadata.yaml:2]: Let's fill in the friendly_name, description, and owners fields here too. We may have logic that requires them, and they're good minimal documentation to give future users context on what this is.

File: sql/moz-fx-data-shared-prod/search_derived/remotesettings_suggestions_v1/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+friendly_name: Remote Settings Suggestions
+description: >
+  A table containing the suggestions importend from Remote Settings,
+  from the same list powering Quicksuggest.
```
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/search_derived/remotesettings_suggestions_v1/metadata.yaml:5]:
```suggestion
  from the same list powering Quicksuggest.
  
  Populated by the `quicksuggest2bq` scheduled task in telemetry-airflow.
```

**Comment by jklukas** [comment]
cc @whd This table is created and populated by a custom task in Airflow, so this metadata is intended primarily to grant appropriate permissions on the table. Does that all sound correct to you?

File: sql/moz-fx-data-shared-prod/search_derived/remotesettings_suggestions_v1/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
```
Comment (Dexterp37) [inline sql/moz-fx-data-shared-prod/search_derived/remotesettings_suggestions_v1/metadata.yaml:6]:
```suggestion

```

**Comment by whd** [comment]
I don't think this is specifically necessary since `search_derived` at the dataset-level should be accessible to `mozilla-confidential`. As far as I can tell, this table doesn't exist yet.

Of note though, we have `search_derived` and `search_terms_derived` datasets, perhaps this is meant for the latter instead of the former.

**Comment by jklukas** [comment]
> I don't think this is specifically necessary since `search_derived` at the dataset-level should be accessible to `mozilla-confidential`. As far as I can tell, this table doesn't exist yet.
> 
> Of note though, we have `search_derived` and `search_terms_derived` datasets, perhaps this is meant for the latter instead of the former.

You are correct! Thank you for catching that!

**Comment by whd** [comment]
> You are correct! Thank you for catching that!

Ok, assuming this is changed to `search_terms_derived`, this looks correct  given what I see in https://mozilla-hub.atlassian.net/browse/ROAD-134?focusedCommentId=482791. We'll likely start enforcing increased change procedures on this dataset next week, which would have some additional procedural overhead (e.g. formal managerial approval).

**Review by jklukas — APPROVED** [review]

**Comment by jklukas** [comment]
I pushed an update to change the path. The Airflow job successfully created the table, so it exists now and I verified that I'm currently unable to access it.

File: sql/moz-fx-data-shared-prod/search_terms_derived/remotesettings_suggestions_v1/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+---
+friendly_name: Remote Settings Suggestions
+description: >
+  A table containing the suggestions importend from Remote Settings,
```
Comment (whd) [inline sql/moz-fx-data-shared-prod/search_terms_derived/remotesettings_suggestions_v1/metadata.yaml:4]:
```suggestion
  A table containing the suggestions imported from Remote Settings,
```

**Review by whd — APPROVED** [review]
I can deploy this once it makes its way to `private-generated-sql`. Since this is a non-restricted table in a restricted dataset, it feels like it would be good to have some note of that in the description or comments. Were I to look at this at some point in the future I'd probably `git log` to see the Jira issue where `workgroup:mozilla-confidential` is noted as the appropriate classification. I'm not sure what classification `Remote Settings` data generally falls under but my assumption is that it's `mozilla-confidential` which is why this can be `mozilla-confidential`.

**Comment by jklukas** [comment]
> I'm not sure what classification Remote Settings data generally falls under but my assumption is that it's mozilla-confidential which is why this can be mozilla-confidential.

Yes, that is correct. Given the coming change control process, it may be worth considering moving it into a non-restricted datasets, even though it's conceptually related to search terms data.

I'll file a follow up with the docs fixups.

File: sql/moz-fx-data-shared-prod/search_terms_derived/remotesettings_suggestions_v1/metadata.yaml (hunk 4 above)
Comment (jklukas) [reply sql/moz-fx-data-shared-prod/search_terms_derived/remotesettings_suggestions_v1/metadata.yaml:4]: Addressed in #2511

**Comment by whd** [comment]
It appears I can't comment on the Jira issue but this ACL change has been applied.

**Pull request merged**